File: src/main/java/com/github/lppedd/cc/lookup/CommitTypeLookupElement.java
package com.github.lppedd.cc.lookup;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import com.github.lppedd.cc.CCIcons;
import com.github.lppedd.cc.psi.CommitTypePsiElement;
import com.intellij.codeInsight.lookup.LookupElement;
import com.intellij.codeInsight.lookup.LookupElementPresentation;
import com.intellij.psi.PsiElement;

public class CommitTypeLookupElement extends LookupElement {
  private final CommitTypePsiElement psiElement;

  public CommitTypeLookupElement(final CommitTypePsiElement psiElement) {
    this.psiElement = psiElement;
  }

  @Override
  public void renderElement(final LookupElementPresentation presentation) {
    presentation.setItemText(psiElement.getCommitTypeName());
    presentation.setIcon(CCIcons.TYPE);
  }

  @Nullable
  @Override
  public PsiElement getPsiElement() {
    return psiElement;
  }

  @NotNull
  @Override
  public String getLookupString() {
    return psiElement.getCommitTypeName();
  }

  @Override
  public boolean isCaseSensitive() {
    return false;
  }
}

File: src/main/kotlin/com/github/lppedd/cc/lookup/CommitTypeLookupElement.kt
package com.github.lppedd.cc.lookup

import com.github.lppedd.cc.ConventionalCommitIcons
import com.github.lppedd.cc.psi.CommitTypePsiElement
import com.intellij.codeInsight.lookup.LookupElementPresentation

/**
 * @author Edoardo Luppi
 */
open class CommitTypeLookupElement(
  override val index: Int,
  private val psiElement: CommitTypePsiElement
) : ConventionalCommitLookupElement() {
  override val weight = 30
  override fun renderElement(presentation: LookupElementPresentation) {
    presentation.run {
      itemText = psiElement.commitTypeName
      icon = ConventionalCommitIcons.TYPE
    }
  }

  override fun getPsiElement() = psiElement
  override fun getLookupString() = psiElement.commitTypeName
}
